fsspec/caching.py

```diff
@@ -93,16 +114,30 @@ def _fetch(self, start, end):
         start_block = start // self.blocksize
         end_block = end // self.blocksize
         need = [i for i in range(start_block, end_block + 1) if i not in self.blocks]
+        ranges = []
         while need:
             # TODO: not a for loop so we can consolidate blocks later to
-            # make fewer fetch calls; this could be parallel
+            # make fewer fetch calls
             i = need.pop(0)
             sstart = i * self.blocksize
             send = min(sstart + self.blocksize, self.size)
-            logger.debug(f"MMap get block #{i} ({sstart}-{send}")
-            self.cache[sstart:send] = self.fetcher(sstart, send)
+            ranges.append((sstart, send))
             self.blocks.add(i)
 
+        if not ranges:
+            return self.cache[start:end]
+
+        if self.multi_fetcher:
```

Comment:
Since .cat_ranges always exists and falls back to getting each block serially, should we just always call that?

Reply:
In `caching.py`, I wanted to maintain backwards compatibility. If someone uses the `MMapCache` class directly, their code would break if we require the `multi_fetcher` style interface.

Reply:
OK, but let's agree to eventually deprecate this